src/app.js

```diff
@@ -0,0 +1,275 @@
+    this.#sequenceStr = ''
+    this.#clipList = []
+    this.#currentOffset = this.#initialOffset
+  }
+  gcd (a, b) {
+    let temp
+    while (b !== 0) {
+      temp = a % b
+      a = b
+      b = temp
+    }
+    return a
+  }
+  toTimecodeStr (frameNumber) {
+    let tcNumerator = frameNumber * this.#tcMultiplier
+    let divisor = this.gcd(tcNumerator, this.#tcDenominator)
+    return (
+      (tcNumerator / divisor).toString() +
+      '/' +
+      (this.#tcDenominator / divisor).toString() +
+      's'
+    )
+  }
+  addClip (startFrame, endFrame) {
+    let asset = ''
+    let clip = ''
+    let assetIDAttr = 'r' + this.#currentAssetID.toString()
+    let assetNameAttr = this.#mediaInfo.fileName
+    let assetDurationAttr = this.toTimecodeStr(this.#mediaInfo.frameCount)
+    let clipStartAttr = this.toTimecodeStr(startFrame)
+    let clipDuration = endFrame - startFrame
+    let clipDurationAttr = this.toTimecodeStr(clipDuration)
+    let clipOffsetAttr = this.toTimecodeStr(this.#currentOffset)
+    if (this.#mediaInfo.hasVideo) {
+      asset = `<asset id="${assetIDAttr}" format="r1" name="${assetNameAttr}" start="0/1s" duration="${assetDurationAttr}" hasVideo="1" hasAudio="1" audioSources="1"><media-rep kind="original-media" src="${assetNameAttr}"/></asset>`
+      clip = `<asset-clip ref="${assetIDAttr}" format="r1" tcFormat="NDF" name="${assetNameAttr}" start="${clipStartAttr}" duration="${clipDurationAttr}" offset="${clipOffsetAttr}" enabled="1"><adjust-transform scale="1 1" anchor="0 0" position="0 0"/></asset-clip>`
+    } else {
+      asset = `<asset id="${assetIDAttr}" name="${assetNameAttr}" start="0/1s" duration="${assetDurationAttr}" hasAudio="1" audioSources="1"><media-rep kind="original-media" src="${assetNameAttr}"/></asset>`
+      clip = `<asset-clip ref="${assetIDAttr}" name="${assetNameAttr}" start="${clipStartAttr}" duration="${clipDurationAttr}" offset="${clipOffsetAttr}" lane="2" enabled="1"/>`
+    }
+    this.#assetList.push(asset)
+    this.#clipList.push(clip)
+    this.#currentAssetID++
+    this.#currentOffset += clipDuration
+  }
+  stringify () {
+    let sequenceTCStartAttr = this.toTimecodeStr(this.#initialOffset)
+    let sequenceDurationAttr = this.toTimecodeStr(
+      this.#currentOffset - this.#initialOffset
+    )
+    if (this.#mediaInfo.hasVideo) {
+      this.#sequenceStr = `<sequence format="r0" tcFormat="NDF" tcStart="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}"><spine>`
+    } else {
+      this.#sequenceStr = `<sequence format="r0" tcFormat="NDF" tcStart="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}"><spine><gap name="Gap" start="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}" offset="${sequenceTCStartAttr}">`
+    }
+    return (
+      this.#startStr +
+      this.#assetList.join('') +
+      this.#eventStr +
+      this.#sequenceStr +
+      this.#clipList.join('') +
+      this.#endStr
+    )
+  }
+} // class FCPXML
```

Comment:
remove // class FCPXML?

Reply:
Ah that was just there to mark the end of the class to make it easy to read, doesn't matter either way as it's just a short inline comment